### PR TITLE
Implement collection tests for release 1.0.0 branch

### PR DESCRIPTION
**Pull request by grantli-os**

### CODE REVIEW

File: tests/references/collections.py [hunk 1]
```diff
@@ -64,153 +88,831 @@
...
+    },
+    {
+        "desc": cleandoc("""Collection with date traits."""),
+        "token_type": "non-fungible",
+        "tokens": {
+            1: {
+                "attributes": [
+                    {
+                        "name": "start",
+                        "value": 1640000000,
```
Comment (theelderbeever) [inline tests/references/collections.py:139]: Could you make a test with string formatted dates (`"2022-01-01T00:00:00"` and `"2022-01-01"`) as well? It should process those as well so knowing that it does would be great.

File: tests/references/collections.py [hunk 2]
```diff
@@ -64,153 +88,831 @@
...
+                    "metric.probability": 0.5,
+                    "metric.information": 1.0,
+                },
+                {
+                    "name": "animated",
+                    "value": "no",
+                    "attribute.token_count": 2,
+                    "attribute.supply": 2,
+                    "metric.probability": 1.0,
+                    "metric.information": -0.0,
```
Comment (theelderbeever) [inline tests/references/collections.py:521]: Any idea what the negative is about? Does the library output like that?

**Review by theelderbeever — CHANGES_REQUESTED** [review]
Please rebase and run `pre-commit` checks. I added an import sorter which affects your code. Otherwise, looks good aside from the couple comments I left.

File: tests/references/collections.py (hunk 1 above)
Comment (grantli-os) [reply tests/references/collections.py:139]: Added

File: tests/references/collections.py (hunk 2 above)
Comment (grantli-os) [reply tests/references/collections.py:574]: The library outputs this, `-log2(attr["attribute.supply"] / total)` will return -0.0 when attribute supply equals total supply.

Comment (theelderbeever) [reply tests/references/collections.py:574]: Ah right right that makes sense. And besides `-0.0 == 0.0` is `True` so nbd.

**Pull request merged**